src/proto.rs

```diff
@@ -0,0 +1,273 @@
+pub struct MeanVarUnsignedVectorClient<'a, F: FieldElement> {
+    phantom: PhantomData<F>,
+    pk: &'a HpkePublicKey,
+    prg: Prg,
+}
+
+impl<'a, F: FieldElement> PrioClient<'a, F, MeanVarUnsigned<F>, MeanVarUnsignedVector<F>>
+    for MeanVarUnsignedVectorClient<'a, F>
+{
+    fn new(param: &PrioParam, pk: &'a HpkePublicKey) -> Result<Self, PrioError> {
+        match param.data_type {
+            PrioDataType::MeanVarUnsignedVector { bits, length } => Ok(Self {
+                phantom: PhantomData::<F>,
+                pk,
+                prg: param.prg,
+            }),
+        }
+    }
+}
+
+pub struct MeanVarUnsignedVectorLeader<F: FieldElement> {
+    phantom: PhantomData<F>,
+    bits: usize,
+    length: usize,
+}
+
+impl<F: FieldElement> PrioLeader<F, MeanVarUnsigned<F>, MeanVarUnsignedVector<F>>
+    for MeanVarUnsignedVectorLeader<F>
+{
+    fn new(param: &PrioParam, leader_share: &[u8]) -> Result<Self, PrioError> {
+        match param.data_type {
+            PrioDataType::MeanVarUnsignedVector { bits, length } => Ok(Self {
+                phantom: PhantomData::<F>,
+                bits,
+                length,
+            }),
+        }
+    }
+}
+
+pub struct MeanVarUnsignedVectorHelper<'a, F: FieldElement> {
+    phantom: PhantomData<F>,
+    sk: &'a HpkeSecretKey,
+    prg: Prg,
+    bits: usize,
+    length: usize,
+}
+
+impl<'a, F: FieldElement> PrioHelper<'a, F, MeanVarUnsigned<F>, MeanVarUnsignedVector<F>>
+    for MeanVarUnsignedVectorHelper<'a, F>
+{
+    fn new(
+        param: &PrioParam,
+        sk: &'a HpkeSecretKey,
+        helper_share: &[u8],
+    ) -> Result<Self, PrioError> {
+        match param.data_type {
+            PrioDataType::MeanVarUnsignedVector { bits, length } => Ok(Self {
+                phantom: PhantomData::<F>,
+                sk,
+                prg: param.prg,
+                bits,
+                length,
+            }),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::field::Field64 as F;
+
+    #[test]
+    fn test_upload_verify() {
+        let buf_len = 1024;
+
+        let pk = HpkePublicKey {}; // XXX
+        let sk = HpkeSecretKey {}; // XXX
+
+        let bits = 12;
+        let measurement = [23, 42, 99, 0, 1, 2, 6, 1337];
+
+        let param = PrioParam {
+            data_type: PrioDataType::MeanVarUnsignedVector {
+                bits,
+                length: measurement.len(),
+            },
+            field: Field::Field64,
+            prg: Prg::Aes128Ctr,
+        };
+
+        // Upload Start: The leader responds to an upload start request by sending the client a
+        // string called `joint_rand_seed`. The aggregators will use this string to verify the
+        // proof.
+        let joint_rand_seed_len = match param.prg {
+            Prg::Aes128Ctr => 16,
+        };
+        let joint_rand_seed: Vec<u8> = (0..joint_rand_seed_len)
+            .map(|_| rand::random::<u8>())
+            .collect();
```

Comment:
Note: `joint_rand_seed` is the client parameters for Prio (i.e., `PrioClientParam`.).